packages/components/fields/text-field/src/text-field.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+  Component,
+  isValidElement,
+  type ChangeEventHandler,
+  type FocusEventHandler,
+  type ReactElement,
+  type ReactNode,
+} from 'react';
+import Constraints from '@commercetools-uikit/constraints';
+import Stack from '@commercetools-uikit/spacings-stack';
+import FieldLabel from '@commercetools-uikit/field-label';
+import TextInput from '@commercetools-uikit/text-input';
+import {
+  filterDataAttributes,
+  createSequentialId,
+  getFieldId,
+  warning,
+} from '@commercetools-uikit/utils';
+import FieldErrors from '@commercetools-uikit/field-errors';
+
+const sequentialId = createSequentialId('text-field-');
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+type TErrorRenderer = (key: string, error?: boolean) => ReactNode;
+
+type TFieldErrors = Record<string, boolean>;
+
+type TTextFieldProps = {
+  // TextField
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint?:
+    | 3
+    | 4
+    | 5
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * A map of errors. Error messages for known errors are rendered automatically.
+   * <br/>
+   * Unknown errors will be forwarded to renderError.
+   */
+  errors?: TFieldErrors;
+  /**
+   * This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   * <br />
+   * Signature: `(key, error) => React.node`
```

Comment:
Nit: this can be removed, as the generator will provide the signature from the type in the readme.